src/main/java/com/openlattice/authorization/Principals.java

```diff
@@ -43,49 +41,26 @@
-    private static final Lock startupLock = new ReentrantLock();
-    private static LoadingCache<String, SecurablePrincipal>      users;
-    private static LoadingCache<String, NavigableSet<Principal>> principals;
+    private static final Logger                                logger      = LoggerFactory
+            .getLogger( Principals.class );
+    private static final Lock                                  startupLock = new ReentrantLock();
+    private static       IMap<String, SecurablePrincipal>      users;
+    private static       IMap<String, NavigableSet<Principal>> principals;
 
     private Principals() {
     }
 
-    public static void init( SecurePrincipalsManager spm ) {
+    public static void init( SecurePrincipalsManager spm, HazelcastInstance hazelcastInstance ) {
         if ( startupLock.tryLock() ) {
-            users = CacheBuilder
-                    .newBuilder()
-                    .expireAfterWrite( 1, TimeUnit.SECONDS )
-                    .build( new CacheLoader<>() {
-                        @Override public SecurablePrincipal load( String principalId ) throws Exception {
-                            return spm.getPrincipal( principalId );
-                        }
-                    } );
-
-            principals = CacheBuilder
-                    .newBuilder()
-                    .expireAfterWrite( 30, TimeUnit.SECONDS )
-                    .build( new CacheLoader<>() {
-                        @Override public NavigableSet<Principal> load( String principalId ) throws Exception {
-                            SecurablePrincipal sp = users.getUnchecked( principalId );
-                            Collection<SecurablePrincipal> securablePrincipals = spm.getAllPrincipals( sp );
-                            if ( securablePrincipals == null ) {
-                                return null;
-                            }
-                            NavigableSet<Principal> currentPrincipals = new TreeSet<>();
-                            currentPrincipals.add( sp.getPrincipal() );
-                            securablePrincipals.stream().map( SecurablePrincipal::getPrincipal )
-                                    .forEach( currentPrincipals::add );
-                            return currentPrincipals;
-                        }
-                    } );
+            users = hazelcastInstance.<String, SecurablePrincipal>getMap( HazelcastMap.PRINCIPALS.name() );
+            principals = hazelcastInstance.<String, NavigableSet<Principal>>getMap( HazelcastMap.PRINCIPALS.name() );
         } else {
             logger.error( "Principals security processing can only be initialized once." );
             throw new IllegalStateException( "Principals context already initialized." );
         }
     }
 
-    public static void requireOrganization( Principal principal ) {
+    public static void ensureOrganization( Principal principal ) {
```

Comment:
I don't think this is used anywhere right now, but could be from `Organization.init`

Reply:
Was not used.